main.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    item_select = []
+    my_each { |i| item_select << i if yield(i) }
+    item_select
+  end
+
+  def my_all?
+    return my_each { |i| return false if yield(i) == false } unless block_given?
+
+    true
+  end
+
+  def my_any?
+    return my_each { |i| return true if yield(i) } unless block_given?
+
+    false
+  end
+
```

Comment:
My any should be able to take arguments please address this, also. You need to handles cases where a block of code is not given.